dipy/tracking/fbcmeasures.pyx

```diff
@@ -0,0 +1,334 @@
+        
+        # estimate which kernel LUT index corresponds to angles
+        tree = KDTree(kernel.get_orientations())
+        for lineId in range(numberOfFibers):
+            for pointId in range(streamlines_length[lineId] - 1):
+                streamlines_nearestp[lineId, pointId] = \
+                    tree.query(streamlines[lineId, pointId])[1]
+        
+        # arrays for parallel computing
+        score_mp = np.zeros(numberOfFibers)
+        xd_mp = np.zeros(numberOfFibers, dtype=np.int32)
+        yd_mp = np.zeros(numberOfFibers, dtype=np.int32)
+        zd_mp = np.zeros(numberOfFibers, dtype=np.int32)
+
+        if have_openmp:
+            print("Running in parallel!")
+        else:
+            print("No OpenMP...")
+
+        # compute fiber LFBC measures
+        with nogil:
+
+            for lineId in prange(numberOfFibers, schedule='guided'):
+                for pointId in range(streamlines_length[lineId] - 1):
+                    score_mp[lineId] = 0.0
+                    for lineId2 in range(numberOfFibers):
+                    
+                        # skip lfbc computation with itself
+                        if lineId == lineId2:
+                            continue
+                            
+                        for pointId2 in range(streamlines_length[lineId2] - 1):
+                            # compute displacement
+                            xd_mp[lineId] = int(streamlines[lineId, pointId, 0] 
+                            - streamlines[lineId2, pointId2, 0] + 0.5)
+                            yd_mp[lineId] = int(streamlines[lineId, pointId, 1] 
+                            - streamlines[lineId2, pointId2, 1] + 0.5)
+                            zd_mp[lineId] = int(streamlines[lineId, pointId, 2] 
+                            - streamlines[lineId2, pointId2, 2] + 0.5)
+                            
+                            # if position is outside the kernel bounds, skip
+                            if xd_mp[lineId] > hn or -xd_mp[lineId] > hn or \
+                               yd_mp[lineId] > hn or -yd_mp[lineId] > hn or \
+                               zd_mp[lineId] > hn or -zd_mp[lineId] > hn:
+                                continue
+                            
+                            # grab kernel value from LUT
+                            score_mp[lineId] += \
+                                        lut[streamlines_nearestp[lineId, pointId], 
+                                        streamlines_nearestp[lineId2, pointId2], 
+                                        hn+xd_mp[lineId], 
+                                        hn+yd_mp[lineId], 
+                                        hn+zd_mp[lineId]]  # ang_v, ang_r, x, y, z
+                                    
+                    streamline_scores[lineId, pointId] = score_mp[lineId]
+
+        if have_openmp and num_threads is not None:
+            openmp.omp_set_num_threads(all_cores)
```

Comment:
This line resets the number of OpenMP threads back to the default (all cores). I will add a comment there.